### PR TITLE
[i18n] translate game code on URL

### PR DESCRIPTION
**Pull request by richardaum**
### Goal

The main goal here is to translate the game code on URLs generated by the Link component and Router functions.

For that reason, the following changes are being made:
- Introducing a new way to determine which URL to navigate on Link `href` property and Router functions: `infra/navigation` methods (play, room, match, home, about). These functions are made to properly get the language and build the URL fully translated (no need `translatedHref` function anymore).
- No need `routing.json` - redirects and rewrites are generated programmatically from the game list.

### CODE REVIEW

**Comment by richardaum** [comment]
I probably still need to work on sitemap, but I think it should be a later work.

File: web/src/games/chess/index.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import instructions from './instructions.md';
 
 export const chessGameDef: IGameDef = {
   code: 'chess',
+  codes: { en: 'chess', pt: 'xadrez' },
```
Comment (vdfdev) [inline web/src/games/chess/index.ts:8]: This looks great!

File: web/src/pages/g/[gameCode]/index.tsx [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
 import { generatePageError } from 'next-with-error';
+import { play } from 'infra/navigation';
+import { getGameDefinition } from 'infra/game';
 
 export default class G extends React.Component {
-  static async getInitialProps({ res, query }) {
+  static async getInitialProps({ req, res, query }) {
     const gameCode = query.gameCode as string;
-    const gameDef: IGameDef = GAMES_MAP[gameCode];
+    const gameDef: IGameDef = getGameDefinition(gameCode);
```
Comment (vdfdev) [inline web/src/pages/g/[gameCode]/index.tsx:11]: Great that we are replacing this global :)

**Comment by vdfdev** [comment]
The code looks great! I am going to manually test here that everything works OK and merge :). Thanks for the amazing work :D

**Comment by vdfdev** [comment]
I am going to check if it works on `master` branch or if it is specific to this branch

**Comment by vdfdev** [comment]
`master` runs OK here, so the issue must be with this branch...
I am going to sleep now, but tomorrow I can help you after lunch here

**Comment by vdfdev** [comment]
Taking a look now :D

**Comment by vdfdev** [comment]
Everything worked now :tada: !!! The only minor bug I found was that if I tried to navigate to a game in the /pt/ that wasnt translated, I got a page not found error... But that does not impact games in english, so we can merge!

**Review by vdfdev — APPROVED** [review]

**Comment by vdfdev** [comment]
Sure! I can send you more details later today

**Pull request merged**